src/editor/export.js

```diff
@@ -7,8 +7,29 @@ const sizeInputs = /** @type {NodeListOf<HTMLInputElement>} */ (document.getElem
   'sizes'
 ));
 
+/** @type {HTMLAnchorElement} */
+const showPreviewArrayIcons = document.querySelector('#showPreviewArrayIcons');
+/** @type {HTMLPreElement} */
+const outputPreview = document.querySelector('#outputPreviewJSON');
+
+let sizesArr = [];
+
 const insightsReady = import('/web_modules/insights-js/dist/esnext/index.js');
 
+showPreviewArrayIcons.addEventListener('click', (e) => {
+  e.preventDefault();
+  const filteredJson = sizesArr
+    .filter((i) => i)
+    .map((size) => ({
+      src: `maskable_icon_x${size}.png`,
+      type: 'image/png',
+      sizes: `${size}x${size}`,
+    }));
+
+  outputPreview.textContent = JSON.stringify(filteredJson, null, 2);
```

Comment:
Updating the text is cheap enough that we can just change it everytime the selected checkboxes are changed. That also removes the need for a separate array, you can read `sizeInputs`.